src/main/java/com/nutomic/syncthingandroid/fragments/DeviceSettingsFragment.java

```diff
@@ -142,6 +149,14 @@ else if (currentState != SyncthingService.State.ACTIVE) {
             }
             mDeviceId.setOnPreferenceClickListener(this);
         }
+
+        if (mDevice == null) {
+            // The device was removed from Syncthing and no reference was found
+            Log.d(TAG, "Empty mDevice");
+            getActivity().finish();
+            return;
+        }
```

Comment:
Also, moving this to a function doesn't help much. We still need to escape/return in the calling function, which involves a similar if test